geode-assembly/src/acceptanceTest/java/org/apache/geode/client/sni/ClientSNIDropProxyAcceptanceTest.java

```diff
@@ -0,0 +1,197 @@
+  @ClassRule
+  public static TestRule ignoreOnWindowsRule = new IgnoreOnWindowsRule();
+
+  @Rule
+  public DockerComposeRule docker = DockerComposeRule.builder()
+      .file(DOCKER_COMPOSE_PATH.getPath())
+      .build();
+
+  private ClientCache cache;
+
+  private String trustStorePath;
+  private int proxyPort;
+
+  @Before
+  public void before() throws IOException, InterruptedException {
+    trustStorePath =
+        createTempFileFromResource(ClientSNIDropProxyAcceptanceTest.class,
+            "geode-config/truststore.jks")
+                .getAbsolutePath();
+    docker.exec(options("-T"), "geode",
+        arguments("gfsh", "run", "--file=/geode/scripts/geode-starter.gfsh"));
+  }
+
+  @After
+  public void after() {
+    ensureCacheClosed();
+  }
+
+  @Test
+  public void performSimpleOperationsDropSNIProxy()
+      throws IOException,
+      InterruptedException {
+    final Region<String, Integer> region = getRegion();
+
+    region.put("Roy Hobbs", 9);
+    assertThat(region.get("Roy Hobbs")).isEqualTo(9);
+
+    stopProxy();
+
+    assertThatThrownBy(() -> region.get("Roy Hobbs"))
+        .isInstanceOf(NoAvailableLocatorsException.class)
+        .hasMessageContaining("Unable to connect to any locators in the list");
+
+
+    restartProxy();
+
+    await().untilAsserted(() -> assertThat(region.get("Roy Hobbs")).isEqualTo(9));
+
+    region.put("Bennie Rodriquez", 30);
+    assertThat(region.get("Bennie Rodriquez")).isEqualTo(30);
+
+    region.put("Jake Taylor", 7);
+    region.put("Crash Davis", 8);
+
+    region.put("Ricky Vaughn", 99);
+    region.put("Ebbie Calvin LaLoosh", 37);
+
+  }
+
+  private void stopProxy() throws IOException, InterruptedException {
+    docker.containers()
+        .container("haproxy")
+        .stop();
+  }
+
+  private void restartProxy() throws IOException, InterruptedException {
+    restartProxyOnPreviousPort();
+    // Leave this commented here in case you need it for troubleshooting
+    // restartProxyOnDockerComposePort();
+  }
+
+  /**
+   * Use this variant to (re)start the container on whatever port(s) is specified in
+   * docker-compose.yml. Usually that would look something like:
+   *
+   * ports:
+   * - "15443:15443"
+   *
+   * Leave this unused method here for troubleshooting.
+   */
+  private void restartProxyOnDockerComposePort() throws IOException, InterruptedException {
+    docker.containers()
+        .container("haproxy")
+        .start();
+  }
+
+  /**
+   * Use this variant to (re)start the container whatever host port it was bound to before
+   * it was stopped. Usually you'll want the ports spec in docker-compose.yml to look like
+   * this when using this method (allowing Docker to initially choose a random host port
+   * to bind to):
+   *
+   * ports:
+   * - "15443"
+   */
+  private void restartProxyOnPreviousPort() throws IOException, InterruptedException {
+    docker.run(
```

Comment:
It would be nice to have a comment describing what the string.format() arguments are about.  Otherwise someone working on this in the future will have to do some digging to figure it out.